File: SwingApps/apps/cc/applets/kaiser/SwingPlayerUser.java
package cc.applets.kaiser;

import cc.game.kaiser.core.Bid;
import cc.game.kaiser.core.Card;
import cc.game.kaiser.core.Kaiser;

public class SwingPlayerUser extends SwingPlayer { 

    KaiserApplet applet;
    
    public SwingPlayerUser() {
        applet = null;
    }
    
    protected SwingPlayerUser(String name, KaiserApplet applet) {
        super(name);
        this.applet = applet;
    }

    @Override
    public boolean isCardsShowing() {
        return true;
    }
    
    @Override
    public void onProcessRound(Kaiser k) {
        synchronized (applet) {
            try {
                applet.wait();
            } catch (Exception e) {}
        }
    }

    @Override
    public Card playTrick(Kaiser kaiser, Card[] options) {
        return applet.pickCard(options);
    }

    @Override
    public Bid makeBid(Kaiser kaiser, Bid[] options) {
        return applet.pickBid(options);
    }
    
}

File: SwingApps/apps/cc/applets/kaiser/SwingPlayerUser.kt
package cc.applets.kaiser

import cc.game.kaiser.core.Bid
import cc.game.kaiser.core.Card
import cc.game.kaiser.core.Kaiser

class SwingPlayerUser : SwingPlayer {
	lateinit var applet: KaiserApplet

	constructor() {
	}

	constructor(name: String, applet: KaiserApplet) : super(name) {
		this.applet = applet
	}

	override fun isCardsShowing(): Boolean {
		return true
	}

	override fun onProcessRound(k: Kaiser) {
	}

	override fun playTrick(kaiser: Kaiser, options: Array<Card>): Card? {
		return applet.pickCard(options)
	}

	override fun makeBid(kaiser: Kaiser, options: Array<Bid>): Bid? {
		return applet.pickBid(options)
	}
}